eth/filters/filter_system_test.go

```diff
@@ -747,23 +747,42 @@ func TestStateChangeSubscription(t *testing.T) {
 				t.Errorf("Test failure: %s: %s", t.Name(), test.description)
 				t.Logf("Actual payload block hash equal expected.\nactual:%+v\nexpected: %+v", actualStateDiff.BlockHash, expectedStateDiff.BlockHash)
 			}
-			for i, e := range expectedStateDiff.UpdatedAccounts {
-				if !bytes.Equal(actualStateDiff.UpdatedAccounts[i].Key, e.Key) {
+
+			// The updated accounts are not necessarily in the same order for the state diffs.
+			// Put the actual in a map for easier lookup.
+			actualUpdatedAccounts := make(map[string]AccountDiff)
+			for _, account := range actualStateDiff.UpdatedAccounts {
+				actualUpdatedAccounts[string(account.Key)] = account
+			}
+
+			for _, e := range expectedStateDiff.UpdatedAccounts {
+				actualUpdatedAccount := actualUpdatedAccounts[string(e.Key)]
+
+				if !bytes.Equal(actualUpdatedAccount.Key, e.Key) {
```

Comment:
Wanna make sure I'm reading this correctly - I _think_ that this assertion is redundant because we're getting the UpdatedAccount from the map above based on it's key, right? I like the idea of leaving it in here, to be explicit, but just wanted to make sure I understood.

Reply:
I thought it was too - but it's not, well not completely. Because"If the requested key doesn't exist, we get the value type's zero value."  So the next value assertion would probably fail, I'd like the key assertion to fail to so it's obvious the key isn't present in the map.